dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2017-2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dirent.h>
+#include <errno.h>
+#include <memory>
+
+#include "dali/core/common.h"
+#include "dali/operators/reader/loader/numpy_loader.h"
+#include "dali/util/file.h"
+#include "dali/operators/reader/loader/utils.h"
+
+
+namespace dali {
+TypeInfo TypeFromNumpyStr(const std::string &format) {
+  if (format == "i8")
+    return TypeInfo::Create<int64_t>();
+  else if (format == "i4")
+    return TypeInfo::Create<int32_t>();
+  else if (format == "f4")
+    return TypeInfo::Create<float>();
+  else if (format == "f8")
+    return TypeInfo::Create<double>();
+  else
+    return TypeInfo();
+}
+
+std::unique_ptr<FileStream> NumpyLoader::ParseHeader(std::unique_ptr<FileStream> file,
+                                                     NumpyParseTarget& target) {
+  // check if the file is actually a numpy file
+  uint8_t* token = new uint8_t[10];
+  int64_t nread = file->Read(token, 10);
+  DALI_ENFORCE(nread == 10, "Can not read header.");
```

Comment:
Make sure that `token` is null terminated. Also I don't see much value in heap allocation here.
```
uint8_t token[11];
int64_t nread = file->Read(token, 10);
DALI_ENFORCE(nread == 10, "Can not read header.");
token[nread] = `\0`;
```

Reply:
should be fixed with next push